### ISSUE TITLE
Accessibility: core-selector is missing semantics

### ISSUE THREAD

**Issue by marcysutton** [Polymer/core-selector#28]
To be accessible, `<core-selector>` needs to communicate its role and state to assistive technologies. It also needs to be accessible from the keyboard, but this issue is focused on semantics.

I don't see this component extending a native `<select>` under the hood and there are no ARIA attributes being added dynamically. Can you elaborate on the lack of accessibility in the current version of `<core-selector>` and if we can expect an update to it? I'm willing to work on a PR, but I don't want to overlap existing efforts.

![Core-selector with missing ARIA attributes](https://cloud.githubusercontent.com/assets/1045233/5240071/bab4fc30-78af-11e4-9d98-49e6d67daaa3.png)

A more accessible version built on-top of core-selector, uses roles, `aria-activedescendant` and child IDs (PR forthcoming):
![Basic-list-box component](https://cloud.githubusercontent.com/assets/1045233/5240076/dd13b0be-78af-11e4-98a3-d64420e8f9f9.png)

If accessibility is meant to be "layered on" as a plugin, it is highly likely developers will forget it...making this component less useful. It really should be built into the core component. Just wondering if that's already on your radar, since I don't see any open issues for it.
